[Tests] Test on linux randomly fails with "Failed: Timeout"

### Error Message
Failed: Timeout >20.0s

### Stacktrace
```python
kwargs = {'channel_seeder': <tribler.core.components.libtorrent.download_manager.download_manager.DownloadManager object at 0x7...r': <tribler.core.components.gigachannel_manager.gigachannel_manager.GigaChannelManager object at 0x7fef7c3d20a0>, ...}
coro = <coroutine object test_channel_update_and_download at 0x7fef7c5da3c0>
task = <Task pending name='Task-6109' coro=<test_channel_update_and_download() running at /home/jenkins/workspace/GH_Tribler_...nkins/.local/lib/python3.8/site-packages/ipv8/taskmanager.py:128, <TaskWakeupMethWrapper object at 0x7fef7c7e5bb0>()]>>

    @functools.wraps(func)
    def inner(**kwargs):
        coro = func(**kwargs)
        if coro is not None:
            task = asyncio.ensure_future(coro, loop=_loop)
            try:
>               _loop.run_until_complete(task)

../../../../.local/lib/python3.8/site-packages/pytest_asyncio/plugin.py:195: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/asyncio/base_events.py:603: in run_until_complete
    self.run_forever()
/usr/lib/python3.8/asyncio/base_events.py:570: in run_forever
    self._run_once()
/usr/lib/python3.8/asyncio/base_events.py:1823: in _run_once
    event_list = self._selector.select(timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <selectors.EpollSelector object at 0x7fef7c365130>, timeout = 0.963

    def select(self, timeout=None):
        if timeout is None:
            timeout = -1
        elif timeout <= 0:
            timeout = 0
        else:
            # epoll_wait() has a resolution of 1 millisecond, round away
            # from zero to wait *at least* timeout seconds.
            timeout = math.ceil(timeout * 1e3) * 1e-3
    
        # epoll_wait() expects `maxevents` to be greater than zero;
        # we want to make sure that `select()` can be called when no
        # FD is registered.
        max_ev = max(len(self._fd_to_key), 1)
    
        ready = []
        try:
>           fd_event_list = self._selector.poll(timeout, max_ev)
E           Failed: Timeout >20.0s

/usr/lib/python3.8/selectors.py:468: Failed
```

[Jenkins console output](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_linux_pytest/3252/testReport/junit/src.tribler.core.components.metadata_store.tests/test_channel_download/test_channel_update_and_download/)
PR: #6821

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,7 +1,8 @@
 [pytest]
-timeout = 5
+timeout = 60
 log_level = INFO
-log_cli_format = %(asctime)s %(levelname)s %(message)s
+log_format = %(asctime)s.%(msecs)03d %(levelname)s %(message)s
+log_cli_format = %(asctime)s.%(msecs)03d %(levelname)s %(message)s
 markers =
     guitest:Tests for GUI. Skipped by default, use --guitests option to enable them
     tunneltest:Slow tests for tunnels. Skipped by default, use --tunneltests option to enable them

--- a/src/tribler/core/components/base.py
+++ b/src/tribler/core/components/base.py
@@ -172,7 +172,9 @@ class Session:
             self._startup_exception = exc
 
     async def shutdown(self):
+        self.logger.info("Stopping components")
         await gather(*[create_task(component.stop()) for component in self.components.values()])
+        self.logger.info("All components are stopped")
 
     def __enter__(self):
         Session._stack.append(self)
@@ -232,10 +234,11 @@ class Component:
         self.started_event.set()
 
     async def stop(self):
-        self.logger.info(f'Stop: {self.__class__.__name__}')
-        self.logger.info("Waiting for other components to release me")
+        component_name = self.__class__.__name__
+        dependants = sorted(component.__class__.__name__ for component in self.reverse_dependencies)
+        self.logger.info(f'Stopping {component_name}: waiting for {dependants} to release it')
         await self.unused_event.wait()
-        self.logger.info("Component free, shutting down")
+        self.logger.info(f"Component {component_name} free, shutting down")
         try:
             await self.shutdown()
         except Exception as e:
@@ -245,7 +248,9 @@ class Component:
             self.stopped = True
             for dep in list(self.dependencies):
                 self._release_instance(dep)
-            self.logger.info("Component free, shutting down")
+            remaining_components = sorted(
+                c.__class__.__name__ for c in self.session.components.values() if not c.stopped)
+            self.logger.info(f"Component {component_name}, stopped. Remaining components: {remaining_components}")
 
     async def run(self):
         pass

--- a/src/tribler/core/components/libtorrent/download_manager/download_manager.py
+++ b/src/tribler/core/components/libtorrent/download_manager/download_manager.py
@@ -35,7 +35,7 @@ from tribler.core.utilities.rest_utils import (
     scheme_from_url,
     url_to_path,
 )
-from tribler.core.utilities.simpledefs import DLSTATUS_SEEDING, MAX_LIBTORRENT_RATE_LIMIT, NTFY, STATEDIR_CHECKPOINT_DIR
+from tribler.core.utilities.simpledefs import DLSTATUS_SEEDING, MAX_LIBTORRENT_RATE_LIMIT, STATEDIR_CHECKPOINT_DIR
 from tribler.core.utilities.unicode import hexlify
 from tribler.core.utilities.utilities import bdecode_compat, has_bep33_support, parse_magnetlink
 from tribler.core.version import version_id
@@ -159,25 +159,30 @@ class DownloadManager(TaskManager):
         self.set_download_states_callback(self.sesscb_states_callback)
 
     def notify_shutdown_state(self, state):
+        self._logger.info(f'Notify shutdown state: {state}')
         self.notifier[notifications.tribler_shutdown_state](state)
 
     async def shutdown(self, timeout=30):
+        self._logger.info('Shutting down...')
         if self.downloads:
+            self._logger.info('Stopping downloads...')
+
             self.notify_shutdown_state("Checkpointing Downloads...")
             await gather(*[download.stop() for download in self.downloads.values()], return_exceptions=True)
             self.notify_shutdown_state("Shutting down Downloads...")
             await gather(*[download.shutdown() for download in self.downloads.values()], return_exceptions=True)
 
-        self.notify_shutdown_state("Shutting down Libtorrent Manager...")
+        self.notify_shutdown_state("Shutting down LibTorrent Manager...")
         # If libtorrent session has pending disk io, wait until timeout (default: 30 seconds) to let it finish.
         # In between ask for session stats to check if state is clean for shutdown.
         # In dummy mode, we immediately shut down the download manager.
         while not self.dummy_mode and not self.is_shutdown_ready() and timeout >= 1:
-            self.notify_shutdown_state("Waiting for Libtorrent to finish...")
+            self.notify_shutdown_state("Waiting for LibTorrent to finish...")
             self.post_session_stats()
             timeout -= 1
             await asyncio.sleep(1)
 
+        self._logger.info('Awaiting shutdown task manager...')
         await self.shutdown_task_manager()
 
         if self.dht_health_manager:
@@ -185,20 +190,20 @@ class DownloadManager(TaskManager):
 
         # Save libtorrent state
         if self.has_session():
+            self._logger.info('Saving state...')
             with open(self.state_dir / LTSTATE_FILENAME, 'wb') as ltstate_file:
                 ltstate_file.write(lt.bencode(self.get_session().save_state()))
 
         if self.has_session() and self.config.upnp:
+            self._logger.info('Stopping upnp...')
             self.get_session().stop_upnp()
-
-        for ltsession in self.ltsessions.values():
-            del ltsession
-        self.ltsessions = None
 
         # Remove metadata temporary directory
         if self.metadata_tmpdir:
+            self._logger.info('Removing temp directory...')
             rmtree(self.metadata_tmpdir, ignore_errors=True)
             self.metadata_tmpdir = None
+        self._logger.info('Shutdown completed')
 
     def is_shutdown_ready(self):
         return all(self.lt_session_shutdown_ready.values())
@@ -401,7 +406,6 @@ class DownloadManager(TaskManager):
             queued_disk_jobs = alert.values['disk.queued_disk_jobs']
             queued_write_bytes = alert.values['disk.queued_write_bytes']
             num_write_jobs = alert.values['disk.num_write_jobs']
-
             if queued_disk_jobs == queued_write_bytes == num_write_jobs == 0:
                 self.lt_session_shutdown_ready[hops] = True
 
@@ -672,10 +676,10 @@ class DownloadManager(TaskManager):
                         'upload_rate_limit': rate}
             self.set_session_settings(lt_session, settings)
 
-    def post_session_stats(self, hops=None):
-        for ltsession in self.ltsessions.values() if hops is None else [self.ltsessions[hops]]:
-            if hasattr(ltsession, "post_session_stats"):
-                ltsession.post_session_stats()
+    def post_session_stats(self):
+        self._logger.info('Post session stats')
+        for session in self.ltsessions.values():
+            session.post_session_stats()
 
     async def remove_download(self, download, remove_content=False, remove_checkpoint=True):
         infohash = download.get_def().get_infohash()

--- a/src/tribler/core/components/libtorrent/libtorrent_component.py
+++ b/src/tribler/core/components/libtorrent/libtorrent_component.py
@@ -40,5 +40,5 @@ class LibtorrentComponent(Component):
     async def shutdown(self):
         await super().shutdown()
         if self.download_manager:
-            self.download_manager.stop_download_states_callback()
+            _ = self.download_manager.stop_download_states_callback()
             await self.download_manager.shutdown()

--- a/src/tribler/core/components/libtorrent/tests/test_download_manager.py
+++ b/src/tribler/core/components/libtorrent/tests/test_download_manager.py
@@ -1,11 +1,9 @@
 from asyncio import Future, gather, get_event_loop, sleep
 from unittest.mock import MagicMock
 
-from ipv8.util import succeed
-
-from libtorrent import bencode
-
 import pytest
+from ipv8.util import succeed
+from libtorrent import bencode
 
 from tribler.core.components.libtorrent.download_manager.download_manager import DownloadManager
 from tribler.core.components.libtorrent.settings import LibtorrentSettings
@@ -354,7 +352,7 @@ async def test_post_session_stats(fake_dlmgr):
     fake_dlmgr.ltsessions[0] = mock_lt_session
 
     # Check for status with session stats alert
-    fake_dlmgr.post_session_stats(hops=0)
+    fake_dlmgr.post_session_stats()
     mock_lt_session.post_session_stats.assert_called_once()
 
 

